website/static/js/osfHelpers.es6.js

```diff
@@ -372,6 +372,50 @@ var decodeText = function(text) {
     return text;
 };
 
+var parseName = function(text) {
+    const suffixList = ['jr','jnr','sr','snr','dr','phd','i','ii','iii','iv','v','md','esq'];
+    let fullName = {
+        firstName: '',
+        middleName: '',
+        lastName: '',
+        suffix: ''
+    };
+
+    text = text.split(/(\s+)/).filter( function(name) {
+        return name.trim().length > 0;
+    });
+
+    if (text.length >= 1) {
```

Comment:
Because each  `if` sub section is doing something very specific for a specific reason in a specific order, it might be nice to have a one liner explaining the logic for maintainability purposes.